edit test cases for findpredicate and others (#259)
AY2223S2-CS2103-W17-2/tp: docs/DeveloperGuide.md

<img src="images/ParserClasses.png" width="600"/>

How the parsing works:

> * When called upon to parse a user command, the `DengueHotspotTrackerParser` class either:
>   * creates an `XYZCommandParser` (`XYZ` is a placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse the user command and create a `XYZCommand` object (e.g., `AddCommand`) which the `DengueHotspotTrackerParser` returns back as a `Command` object.
>   * or creates an `XYZCommand`.

Reviewer: ```suggestion
* When called upon to parse a user command, the `DengueHotspotTrackerParser` class will do one of the following:
  * create an `XYZCommandParser` (`XYZ` is a placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse the user command and create a `XYZCommand` object (e.g., `AddCommand`) that the `DengueHotspotTrackerParser` returns back as a `Command` object.
  * create an `XYZCommand`.
```

To avoid the weird either/or split in the bullet points!

    * Cons: Can be very messy to implement.

> ### Data archiving

Reviewer: ```suggestion
### Data import and export
```
Think "archiving" is a bit different from "saving offline"—with "archiving" I would expect archived cases to still be accessible somewhere in the app?
